test/tests/userobjects/element_subdomain_modifier/block_restricted.i

```diff
@@ -0,0 +1,70 @@
+[Problem]
+  solve = false
+[]
+
+[Mesh]
+  [gen]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 16
+    ny = 16
+  []
+  [left]
+    type = SubdomainBoundingBoxGenerator
+    input = 'gen'
+    block_id = 1
+    bottom_left = '0 0 0'
+    top_right = '0.25 1 1'
+  []
+  [right]
+    type = SubdomainBoundingBoxGenerator
+    input = 'left'
+    block_id = 2
+    bottom_left = '0.25 0 0'
+    top_right = '1 1 1'
+  []
+[]
+
+[UserObjects]
+  [moving_circle]
+    type = CoupledVarThresholdElementSubdomainModifier
+    coupled_var = 'phi'
+    block = 2
+    criterion_type = BELOW
+    threshold = 0
+    subdomain_id = 1
+    moving_boundary_name = moving_boundary
+    execute_on = 'INITIAL TIMESTEP_BEGIN'
```

Comment:
There is still a bug I need to fix: if I don't add INITIAL here, the "moving_boundary" nodeset/sideset won't be created.

Reply:
Update: this is due to the current limitation that libmesh doesn't output empty nodeset/sideset. This will be fixed after [https://github.com/libMesh/libmesh/pull/2871](https://github.com/libMesh/libmesh/pull/2871).

Reply:
Do you plan on waiting for a libmesh update to get this in or do you want to get it in first? If the latter - make an issue and be sure to reference the issue on the test.

Reply:
There is no time pressure on this, so I'm fine either way. I prefer to get this in first (and a regression test for the empty nodeset/sideset after a libmesh update). But I'm open to the other option if you think it is better.

Reply:
There is also a possibility that a libmesh update comes before this PR is ready to merge.

Reply:
I have no preference. Just if you choose to get it in before the libmesh update, please create an issue and reference it in the test/test spec

Reply:
Okay I created an issue and referenced it.